packages/apollo-link-schema/src/schemaLink.ts

```diff
@@ -0,0 +1,57 @@
+import { ApolloLink, Operation, FetchResult, Observable } from 'apollo-link';
+
+import { print } from 'graphql/language/printer';
+import { graphql, GraphQLSchema } from 'graphql';
+
+export class SchemaLink extends ApolloLink {
+  public schema: GraphQLSchema;
+  public rootValue: any;
+  public context: any;
+
+  constructor({
+    schema,
+    rootValue,
+    context
+  }: {
+    schema: GraphQLSchema;
+    rootValue?: any;
+    context?: any;
+  }) {
+    super();
+
+    this.schema = schema;
+    this.rootValue = rootValue;
+    this.context = context;
+  }
+
+  public request(operation: Operation): Observable<FetchResult> | null {
+    const request = {
+      ...operation,
+      query: print(operation.query)
+    };
+
+    return new Observable<FetchResult>(observer => {
+      graphql(
+        this.schema,
+        request.query,
```

Comment:
@jbaxleyiii Hiya, I've written a small addition to the documentation a couple of days ago which coincidentally has been replaced with this.

But I've noticed that you're restringifying the query here instead of using `graphql/execute` and passing through the query AST, which I think can be avoided :smile: (see my docs change) https://github.com/philpl/apollo-client/blame/82a145ef4909a9433ebf3a8c5ef627ceddde79ae/docs/source/recipes/server-side-rendering.md#L233-L236

Reply:
Sorry about stepping on your toes there! Good catch though. I'll put in another PR to correct that.

Reply:
Someone beat me to it. xD

Reply:
Yea, no worries :wink: just wanted to let you know. @imranolas is actually my colleague at Formidable who's having a look at Apollo Link and the Apollo Docs too currently 😬

Reply:
Ahh, formidable. <3 you folks.